File: backend/app/schemas/conversation.py
from enum import Enum

from pydantic import BaseModel, Field


class Role(str, Enum):
    USER = "user"
    ASSISTANT = "assistant"


class Message(BaseModel):
    role: Role
    content: str


class ChatRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=2000)
    history: list[Message] = Field(default_factory=list)


class ChatResponse(BaseModel):
    role: Role = Role.ASSISTANT
    content: str

File: mobile/types/conversation.ts
export enum Role {
  USER = "user",
  ASSISTANT = "assistant",
}

export interface Message {
  id: string;
  role: Role;
  content: string;
  timestamp: number;
}

export interface ConversationRequest {
  message: string;
  history: { role: Role; content: string }[];
}

export interface ConversationResponse {
  role: Role;
  content: string;
}

export interface ConversationState {
  messages: Message[];
  isLoading: boolean;
  error: string | null;
}

export type ConversationAction =
  | { type: "ADD_MESSAGE"; message: Message }
  | { type: "SET_LOADING"; isLoading: boolean }
  | { type: "SET_ERROR"; error: string | null }
  | { type: "LOAD"; messages: Message[] }
  | { type: "CLEAR" };
